src/mono/wasm/debugger/DebuggerTestSuite/AssignmentTests.cs

```diff
@@ -0,0 +1,238 @@
+                (locals) => CheckString(locals, "r", default),
+                (locals) => CheckString(locals, "r", "hello")
+                );
+        }
+
+        [Fact]
+        async Task InspectEnumAssignmentsDuringSteppingIn()
+        {
+            await InspectAssignmentDuringSteppingIn("MONO_TYPE_ENUM",
+                (locals) => CheckEnum(locals, "r", "DebuggerTests.RGB", "Red"),
+                (locals) => CheckEnum(locals, "r", "DebuggerTests.RGB", "Blue")
+                );
+        }
+
+        [Fact]
+        async Task InspectArrayAssignmentsDuringSteppingIn()
+        {
+            await InspectAssignmentDuringSteppingIn("MONO_TYPE_ARRAY",
+                (locals) => CheckObject(locals, "r", "byte[]", is_null: true),
+                (locals) => CheckArray(locals, "r", "byte[]", 2)
+                );
+        }
+
+        [Fact]
+        async Task InspectStructAssignmentsDuringSteppingIn()
+        {
+            await InspectAssignmentDuringSteppingIn("MONO_TYPE_VALUETYPE",
+                (locals) => CheckValueType(locals, "r", "DebuggerTests.Point"),
+                (locals) => CheckValueType(locals, "r", "DebuggerTests.Point")
+                );
+        }
+
+        [Fact]
+        async Task InspectDecimalAssignmentsDuringSteppingIn()
+        {
+            await InspectAssignmentDuringSteppingIn("MONO_TYPE_VALUETYPE2",
+                (locals) => CheckValueType(locals, "r", "System.Decimal", description: "0"),
+                (locals) => CheckValueType(locals, "r", "System.Decimal", description: "1.1")
+                );
+        }
+
+        [Fact]
+        async Task InspectFunctionAssignmentsDuringSteppingIn()
+        {
+            await InspectAssignmentDuringSteppingIn("MONO_TYPE_GENERICINST",
+                (locals) => CheckObject(locals, "r", "System.Func<int>", is_null: true),
+                (locals) => CheckObject(locals, "r", "System.Func<int>", description: "int Prepare ()")
+                );
+        }
+
+        [Fact]
+        async Task InspectFunctionPtrAssignmentsDuringSteppingIn()
+        {
+            await InspectAssignmentDuringSteppingIn("MONO_TYPE_FNPTR",
+                (locals) => CheckSymbol(locals, "r", "(*()) 0"),
+                (locals) => { } // CheckObject(locals, "r", "(*()) 0x22dd230") // FixMe: Could we do better ?
+                );
+        }
+
+        [Fact]
+        async Task InspectIntPtrAssignmentsDuringSteppingIn()
+        {
+            await InspectAssignmentDuringSteppingIn("MONO_TYPE_PTR",
+                (locals) => CheckSymbol(locals, "r", "(int*) 0"),
+                (locals) => { } //CheckObject(locals, "r", "(int*) 0x211c058") // FixMe: Could we do better ?
+                );
+        }
+
+        private async Task InspectAssignmentDuringSteppingIn(string clazz, Action<JToken> checkDefault, Action<JToken> checkValue)
+        {
+            await SetBreakpointInMethod("debugger-test", "DebuggerTests." + clazz, "Prepare", 2);
```

Comment:
nit: Breakpoint can be set directly in the `TestedMethod`, `offset=0`, thus avoiding potential influence of *stepping* into `TestedMethod`. (maybe a bit paranoid ;)

Reply:
Some of the tests have custom TestedMethod for them, which would require custom breakpoint location and complicate stuff. I will keep it as is.